weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallPage/ObjectViewer.tsx

```diff
@@ -92,6 +111,12 @@ export const ObjectViewer = ({
     setExpandedRefs(eRefs => ({...eRefs, [path]: ref}));
   }, []);
 
+  useEffect(() => {
```

Comment:
nit: i would put this, and 
```
  const [truncatedData, setTruncatedData] = useState<Data>(
    traverseAndTruncate(data).result
  );

  const [truncatedStore, setTruncatedStore] = useState<TruncatedStore>({});
```

all in a hook that can be called once in the component